File: android/src/oldarch/com/reactnativecommunity/imageeditor/ImageEditorModule.java
package com.reactnativecommunity.imageeditor;

import java.util.Collections;
import java.util.Map;

import com.facebook.react.bridge.Promise;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactContextBaseJavaModule;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.ReadableMap;
import com.facebook.react.module.annotations.ReactModule;

@ReactModule(name = ImageEditorModule.NAME)
public class ImageEditorModule extends ReactContextBaseJavaModule {
  private ImageEditorModuleImpl moduleImpl;

  public ImageEditorModule(ReactApplicationContext reactContext) {
    super(reactContext);
    moduleImpl = new ImageEditorModuleImpl(reactContext);
  }

  public static final String NAME = ImageEditorModuleImpl.NAME;

  @Override
  public String getName() {
    return ImageEditorModuleImpl.NAME;
  }


  @Override
  public Map<String, Object> getConstants() {
    return Collections.emptyMap();
  }

  @Override
  public void onCatalystInstanceDestroy() {
    moduleImpl.onCatalystInstanceDestroy();
  }

  @ReactMethod
  public void cropImage(String uri, ReadableMap options, Promise promise) {
    moduleImpl.cropImage(uri, options, promise);
  }
}

File: android/src/oldarch/com/reactnativecommunity/imageeditor/ImageEditorModule.kt
package com.reactnativecommunity.imageeditor

import com.facebook.imagepipeline.core.ImagePipeline
import com.facebook.react.bridge.Promise
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.bridge.ReactContextBaseJavaModule
import com.facebook.react.bridge.ReactMethod
import com.facebook.react.bridge.ReadableMap
import com.facebook.react.module.annotations.ReactModule
import com.facebook.react.views.image.ReactCallerContextFactory

@ReactModule(name = ImageEditorModule.NAME)
class ImageEditorModule : ReactContextBaseJavaModule {
    private val moduleImpl: ImageEditorModuleImpl

    constructor(reactContext: ReactApplicationContext) : super(reactContext) {
        moduleImpl = ImageEditorModuleImpl(reactContext, this, null, null)
    }

    constructor(reactContext: ReactApplicationContext, callerContext: Any?) : super(reactContext) {
        moduleImpl = ImageEditorModuleImpl(reactContext, callerContext, null, null)
    }

    constructor(
        reactContext: ReactApplicationContext,
        imagePipeline: ImagePipeline?,
        callerContextFactory: ReactCallerContextFactory?
    ) : super(reactContext) {
        moduleImpl = ImageEditorModuleImpl(reactContext, null, callerContextFactory, imagePipeline)
    }

    override fun getName(): String {
        return ImageEditorModuleImpl.NAME
    }

    override fun invalidate() {
        moduleImpl.invalidate()
        super.invalidate()
    }

    @ReactMethod
    fun cropImage(uri: String, options: ReadableMap, promise: Promise) {
        moduleImpl.cropImage(uri, options, promise)
    }

    companion object {
        const val NAME = ImageEditorModuleImpl.NAME
    }
}
